packages/app/src/app/pages/Dashboard/Sidebar/TemplateItem.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FunctionComponent } from 'react';
+import {
+  DropTarget,
+  DropTargetCollector,
+  DropTargetConnector,
+  DropTargetMonitor,
+  DropTargetSpec,
+} from 'react-dnd';
+
+import { MAKE_TEMPLATE_DROP_KEY } from '../Content/SandboxCard';
+
+import { Item } from './Item';
+
+import TemplateIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/template.svg';
+
+type ComponentProps = {
+  currentPath: string;
+  teamId?: string;
+};
+type Props = ComponentProps & CollectedProps;
+const TemplateItemComponent: FunctionComponent<Props> = ({
+  canDrop,
+  connectDropTarget,
+  currentPath,
+  isOver,
+  teamId,
+}) => {
+  const path = teamId
+    ? `/dashboard/teams/${teamId}/templates`
+    : `/dashboard/templates`;
+
+  return connectDropTarget(
+    <Item
+      active={currentPath === path}
+      Icon={TemplateIcon}
+      name={teamId ? 'Team Templates' : 'My Templates'}
+      path={path}
+      style={isOver && canDrop ? { backgroundColor: 'rgba(0, 0, 0, 0.3)' } : {}}
+    />
+  );
+};
+
+const entryTarget: DropTargetSpec<ComponentProps> = {
+  canDrop: (_props, { getItem }) => getItem() !== null,
+  drop: ({ teamId }, { isOver }) => {
+    // Check if only child is selected:
+    if (!isOver({ shallow: true })) {
+      return {};
+    }
+
+    // Used in SandboxCard
+    return {
+      [MAKE_TEMPLATE_DROP_KEY]: true,
+      teamId,
+    };
+  },
+};
+
+const collectTarget: DropTargetCollector<CollectedProps> = (
+  { dropTarget },
+  { canDrop, isOver }
```

Comment:
You still have to use `connectMonitor` and `monitor` here. The reason is that internally they use `this`. When you destructure, the functions looses their `this` context 😄